src/radical/pilot/agent/agent_0.py

```diff
@@ -108,7 +118,8 @@ def __init__(self, cfg, session):
                                 term_cb=self._hb_term_cb,
                                 log=self._log)
         self._hb.start()
-
+        # Event to handle for services started
+        self.services_event = threading.Event()
```

Comment:
It makes sense to define this in the constructor - but why is it mixed into the hearbeat code?  Please keep related code together and move it up to the service state cb.

Reply:
Woops. I see now. I dont know why I did that specifically.